app/code/community/Bolt/Boltpay/Controller/Traits/ApiControllerTrait.php

```diff
@@ -0,0 +1,109 @@
+<?php
+/**
+ * Bolt magento plugin
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://opensource.org/licenses/osl-3.0.php
+ *
+ * @category   Bolt
+ * @package    Bolt_Boltpay
+ * @copyright  Copyright (c) 2018 Bolt Financial, Inc (https://www.bolt.com)
+ * @license    http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+/**
+ * Trait Bolt_Boltpay_Controller_Traits_ApiControllerTrait
+ *
+ * Defines generalized actions associated with API calls to and from the Bolt server
+ *
+ * @method Mage_Core_Controller_Response_Http getResponse()
+ * @method Mage_Core_Model_Layout getLayout()
+ */
+trait Bolt_Boltpay_Controller_Traits_ApiControllerTrait {
+    use Bolt_Boltpay_BoltGlobalTrait;
+    /**
+     * @var string The body of the request made to this controller
+     */
+    protected $payload;
+    /**
+     * @var string The signed payload which used the stores signing secret
+     */
+    protected $signature;
```

Comment:
I don't think this has to be instance variable. you can inline this within function

Reply:
If this was commonly derived for an HMAC header and was always set, then I would follow that approach and just pull this in the verification method.  However, since there are distinct instances where the signature is not in the HMAC header, (e.g. in the success page the signature is encoded as  a URL get parameter), it is much more elegant to have this as an instance variable accessible throughout the inheriting classes.

Reply:
I was blind, but now I see.  signature removed as class data member.